src/dune_rules/odoc.ml

```diff
@@ -304,13 +373,48 @@ let setup_library_odoc_rules cctx (library : Library.t) ~dep_graphs =
   let modules_and_odoc_files =
     Modules.fold_no_vlib modules ~init:[] ~f:(fun m acc ->
         let compiled =
-          compile_module sctx ~includes ~dep_graphs ~obj_dir ~pkg_or_lnu m
+          compile_module_to_odoc sctx ~includes ~dep_graphs ~obj_dir ~pkg_or_lnu
+            m
         in
         compiled :: acc)
   in
   Dep.setup_deps ctx (Lib local_lib)
     (Path.Set.of_list_map modules_and_odoc_files ~f:(fun (_, p) -> Path.build p))
 
+let setup_library_odocl_rules cctx (library : Library.t) ~dep_graphs =
+  let lib =
```

Comment:
I don't think this function is necessary.

Reply:
We need this specially for `.odocl` rules. I confess much of the code is the same as that in `setup_library_odocl_rules` thus I'll need to abstract it in one